abi/abi.go

```diff
@@ -150,46 +150,58 @@ func (item *Item) Done() {
 // Decodes ABI encoded bytes into an [Item] according to
 // the 'schema' defined by t. For example:
 //	Decode(b, schema.Tuple(schema.Dynamic(), schema.Static()))
-func Decode(input []byte, t schema.Type) *Item {
+// Returns the item and the number of bytes read from input
+func Decode(input []byte, t schema.Type) (int, *Item) {
```

Comment:
bit of drive-by feedback: I feel like it makes more sense from a Go standpoint to return (*Item, int). Felt weird writing code against this that had it the other way around.

Reply:
Is there Go code you cant point to where n is not first? Here are some examples of where it is first:

- https://pkg.go.dev/bytes#Buffer.Read
- https://pkg.go.dev/io#Reader

Reply:
Yeah, totally hear you there. My rationale is that I feel like `*Item` is the primary / most relevant value that decode is returning, not size.

Even in one of the packages you shared, `ReadRune` is similar to what I mean:

* https://pkg.go.dev/bytes#Buffer.ReadRune

Reply:
I also think that `Buffer.Read` is potentially a bad example because it mutates the pointer its on. If it was a function that actually returned a value (e.g. the buffer or similar), I bet that would be the first returned parameter, not the size.

Reply:
Feel free to disagree, just wanted to make my case :)